srcs/environment/set_dup_key_value.c

```diff
@@ -0,0 +1,28 @@
+#include "minishell.h"
+#include "ft_string.h"
+#include "ft_mem.h"
+
+void	env_set_dup_key_value(t_env *env, \
+								const char *key, \
+								const char *value, \
+								t_env_op op)
+{
+	char	*dup_key;
+	char	*dup_value;
+
+	dup_key = NULL;
+	if (key)
+	{
+		dup_key = ft_strdup(key);
+		if (!dup_key)
+			ft_abort();
+	}
```

Comment:
* 環境変数として`!key`なるものは存在しないので、`!key`のケースは許容しなくても良い気がします...！
* 引数に`key = NULL`が来るケースもないとは思いますが、
`if (!key); return;` などのハンドリング要否についても相談しておきたいです！
  * env_ prefix の関数なので valid env keyのみ登録したい
  * 非staticな関数化しているので、keyにNULL含め非valid env keyを入れられるケースはありそう
    * そう考えると `env_set_dup_key_value()` 内で `validate_env_key()` を呼ぶ必要がありそう... ?
    * でもそれは過剰な気もするし、重複して検査するケースが多くなる...
```suggestion
	dup_key = ft_strdup(key);
	if (!dup_key)
		ft_abort();
```